src/modals/storage-connections/StorageConnectionsModal.js

```diff
@@ -0,0 +1,154 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import PluginApiModal from '_/components/modals/PluginApiModal'
+import getPluginApi from '_/plugin-api'
+import { msg } from '_/intl-messages'
+import { Spinner } from 'patternfly-react'
+import { Button } from '@patternfly/react-core'
+import StorageConnectionsModalBody from './StorageConnectionsModalBody'
+import { createErrorMessage } from '_/utils/error-message'
+import './connections.css'
+
+const StorageConnectionsModal = ({
+  isLoading = false,
+  storageDomain,
+  domainConnections,
+  allConnections,
+  onClose = () => { },
+  doConnectionCreate = () => { },
+  doConnectionEdit = () => { },
+  doConnectionDelete = () => { },
+  doConnectionAttach = () => { },
+  doConnectionDetach = () => { },
+  doRefreshConnections = () => { },
+}) => {
+  const [isOpen, setOpen] = useState(true)
+  const [error, setError] = useState(null)
+  const [isShowAll, setShowAll] = useState(false)
+
+  const warning = (!storageDomain || storageDomain.status === 'maintenance')
+    ? null
+    : {
+      title: msg.storageConnectionsDomainNotInMaintenanceWarning(),
+      detail: msg.storageConnectionsDomainNotInMaintenanceWarningDetail(),
+    }
+
+  const close = () => {
+    setOpen(false)
+    onClose()
+  }
+
+  const modalActions = [
+    <Button
+      key='storage-connections-modal-cancel-button'
+      variant='primary'
+      onClick={close}
+    >
+      {msg.closeButton()}
+    </Button>,
+  ]
+
+  const createTitle = () => {
+    let title = msg.storageConnectionsTitle()
+    if (storageDomain) {
+      title = title + ' - ' + storageDomain.name
+    }
+    return title
+  }
+
+  const onConnectionCreate = async (connection) => {
```

Comment:
@sjd78 `async` function as a callback looks risky to me, what do you think?

Reply:
@sjd78 everything is OK as long as you don't call methods tied to React lifecycle i.e. things like `setShowAll(true)` are IMHO allowed only inside synchronous callback.

Reply:
@rszwajko - It is basically the same as a `useEffect()` function, and we're ok using state setters in. An async call in a useEffect() could finish after the component is unmounted, same as these.  I don't consider it much of a risk at this point.